src/main/java/org/example/bookstore/controller/BookControllerImpl.java

```diff
@@ -18,41 +21,51 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
-@Component
 @RestController
-@AllArgsConstructor
+@RequiredArgsConstructor
 @RequestMapping(value = "/api/books")
+@Tag(name = "Book Controller", description = "This controller handles requests "
+        + "and responses related to books in the database")
 public class BookControllerImpl {
 
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> findAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get all existing books from DB")
+    public List<BookDto> getAllBooks(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Get existing book by id")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody BookRequestDto requestDto) {
-        return bookService.save(requestDto);
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create book", description = "Create book with parameters in DB")
+    public BookDto createBook(@RequestBody @Valid BookRequestDto requestDto) {
+        return bookService.create(requestDto);
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBookById(@PathVariable Long id,@RequestBody BookRequestDto book) {
-        return bookService.updateBookById(id,book);
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @Operation(summary = "Update book by id", description = "Update existing book "
+            + "by id with valid parameters")
+    public BookDto updateBookById(@PathVariable Long id, @Valid @RequestBody BookRequestDto book) {
+        return bookService.updateBookById(id, book);
     }
 
     @GetMapping("/search")
-    public List<BookDto> search(BookSearchParameters searchParameters) {
-        return bookService.search(searchParameters);
+    public List<BookDto> search(BookSearchParameters searchParameters,Pageable pageable) {
```

Comment:
* This method is missing a `@Operation`. 
* BookSearchParameters could also have some description of its fields